vertical-pod-autoscaler/pkg/updater/logic/updater.go

```diff
@@ -201,29 +222,60 @@ func (u *updater) RunOnce(ctx context.Context) {
 	vpasWithEvictablePodsCounter := metrics_updater.NewVpasWithEvictablePodsCounter()
 	vpasWithEvictedPodsCounter := metrics_updater.NewVpasWithEvictedPodsCounter()
 
+	vpasWithInPlaceUpdatablePodsCounter := metrics_updater.NewVpasWithInPlaceUpdtateablePodsCounter()
+	vpasWithInPlaceUpdatedPodsCounter := metrics_updater.NewVpasWithInPlaceUpdtatedPodsCounter()
+
 	// using defer to protect against 'return' after evictionRateLimiter.Wait
 	defer controlledPodsCounter.Observe()
 	defer evictablePodsCounter.Observe()
 	defer vpasWithEvictablePodsCounter.Observe()
 	defer vpasWithEvictedPodsCounter.Observe()
+	// separate counters for in-place
+	defer vpasWithInPlaceUpdatablePodsCounter.Observe()
+	defer vpasWithInPlaceUpdatedPodsCounter.Observe()
 
 	// NOTE: this loop assumes that controlledPods are filtered
-	// to contain only Pods controlled by a VPA in auto or recreate mode
+	// to contain only Pods controlled by a VPA in auto, recreate, or inPlaceOrRecreate mode
 	for vpa, livePods := range controlledPods {
 		vpaSize := len(livePods)
 		controlledPodsCounter.Add(vpaSize, vpaSize)
-		evictionLimiter := u.evictionFactory.NewPodsEvictionRestriction(livePods, vpa)
-		podsForUpdate := u.getPodsUpdateOrder(filterNonEvictablePods(livePods, evictionLimiter), vpa)
+		evictionLimiter := u.evictionFactory.NewPodsEvictionRestriction(livePods, vpa, u.patchCalculators)
+		podsForUpdate := u.getPodsUpdateOrder(filterNonUpdatablePods(livePods, evictionLimiter), vpa)
```

Comment:
Would it make things cleaner if we split out the pods that can be in-place updated?

```
podsForInPlaceUpdates := u.getPodsUpdateOrder(filterNonUpdatablePods(livePods, evictionLimiter), vpa)
podsForEviction := u.getPodsUpdateOrder(filterNonEvictablePods(livePods, evictionLimiter), vpa)
```

Then below you can treat them as two different cases with two different for-loops.

Reply:
I've refactored it a bit, lmk if it makes sense.